src/dotnet/ToolPackage/ToolPackageInstaller.cs

```diff
@@ -44,6 +44,10 @@ internal class ToolPackageInstaller : IToolPackageInstaller
                 throw new ArgumentNullException(nameof(packageId));
             }
 
+            // Normalize the package identifier to lower case
+            // This matches the behavior of NuGet
+            packageId = packageId.ToLower();
```

Comment:
A simple string is not enough then. We need to warp it with a type. Later people will compare it and we will have a bug. Maybe there is a class from nuget. If it is simple enough we can take it

Reply:
This should be ToLowerInvariant(). Normalization shouldn't depend on CurrentCulture.

Reply:
Agreed.  I attempted to replicate what NuGet client was doing to package ids, but now that I look again, the `ToLower` calls were in a few tests only (GitHub's search is...not great).  The actual client code is calling `ToLowerInvariant` as expected.  I will fix this and the other `ToLower` calls.

Reply:
Also, I'm fine with encapsulating the id in a type (perhaps along with the version) that normalizes the ids.

Reply:
FYI: There is a PackageIdentity class in NuGet that uses ordinal, case-insensitive comparison on the name. Perhaps we can leverage that instead of making an extra type?

Reply:
Perfect.  We'll use that.